tests/environments/test_tic_tac_toe_engine.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+from unittest import TestCase
+from itertools import product
+
+from environments.tic_tac_toe.tic_tac_toe_engine import TicTacToeEngine
+from environments.tic_tac_toe.tic_tac_toe_engine_utils import Player, Winning, IllegalMoveError, TicTacToeAction, \
+    TicTacToeActionSpace
+
+
+class TestTicTacToeEngine(TestCase):
+    def setUp(self):
+        self.no_of_players = 3
+        self.board_size = 5
+        self.marks_required = 3
```

Comment:
Same as above

Reply:
It is tested in testTicTacToeAdditional.py

Reply:
So you're saying that you have less test cases than dimensions in space (players,size,marks)... It's not extensive testing.

Reply:
So we should hire some students to test our code extensively